apps/frontend/src/app/Components/Conditional/ConditionalSelector.tsx

```diff
@@ -111,16 +119,23 @@ function ExclusiveConditionalSelector({
   )
 
   const conditionalValue = data.get(conditional.value).value
-  const state = conditionalValue
-    ? conditional.states[conditionalValue]
-    : undefined
+  const condStates = evalIfFunc(conditional.states, data)
+  const state = conditionalValue ? condStates[conditionalValue] : undefined
   const badge = state ? (
     getStateBadge(state.name)
   ) : (
     <SqBadge color="secondary">Not Active</SqBadge>
   )
   const condName = getCondName(conditional.name)
 
+  // Conditional value is present, but there is no corresponding state.
+  // Can be caused by old db, or by changing constellation
+  // or changing a condition that disables another state
+  if (!state && conditionalValue) {
+    // Clear out the value
+    setConditional(undefined)
```

Comment:
This is "UI data validation", and something that I dislike happening, because it means that there is a point where the data is "invalid" in the database side, and you'd need to render the UI to correct it. In my opinion, I much prefer to wrap the conditional data in a "range" node, so that validation is actually built into the system itself. I understand that not to be possible in WR, so this is another design consideration for Pando Conditionals.

Reply:
We can do both if you want. I didn't add the range node because it was janky in UI rendering. But with UI validation, that means we can fix the UI side and also not require the UI validation to fix a bad db, until the user loads the UI anyways

Reply:
Removed UI data validation and opted for validation on the sheet side, with visual fixes to the selector. Also fixed a bug where I was changing state from cross component, causing error on react